yt_dlp/extractor/ilpost.py

```diff
@@ -23,46 +23,85 @@ class IlPostIE(InfoExtractor):
-            endpoint_metadata['ajax_url'], display_id, data=urlencode_postdata({
-                'action': 'checkpodcast',
-                'cookie': endpoint_metadata['cookie'],
-                'post_id': episode_id,
-                'podcast_id': podcast_id,
-            }))
-
-        episode = traverse_obj(podcast_metadata, (
-            'data', 'postcastList', lambda _, v: str(v['id']) == episode_id, {dict}), get_all=False)
-        if not episode:
-            raise ExtractorError('Episode could not be extracted')
+        try:
+            episode = self._search_nextjs_data(webpage, display_id)['props']['pageProps']['data']['data']['episode']['data'][0]
+            episode_id = episode['id']
+        except KeyError:
+            raise ExtractorError('Failed to extract episode')
 
         return {
-            'id': episode_id,
+            'id': str(episode_id),
             'display_id': display_id,
-            'series_id': podcast_id,
-            'vcodec': 'none',
             **traverse_obj(episode, {
-                'title': ('title', {str}),
-                'description': ('description', {str}),
-                'url': ('podcast_raw_url', {url_or_none}),
+                'series_id': ('parent', 'id', {int_or_none}),
+                'title': ('title', {unescapeHTML}),
+                'description': ('content_html', {clean_html}),
+                'url': ('episode_raw_url', {url_or_none}),
                 'thumbnail': ('image', {url_or_none}),
                 'timestamp': ('timestamp', {int_or_none}),
                 'duration': ('milliseconds', {float_or_none(scale=1000)}),
-                'availability': ('free', {lambda v: 'public' if v else 'subscriber_only'}),
+                'availability': ('access_level', {lambda v: 'public' if v == 'all' else 'subscriber_only'}),
             }),
         }
+
+
+class IlPostPodcastIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?ilpost\.it/podcasts/(?P<id>[\w\-]+)[/?#]?$'
+    _TESTS = [{
+        'url': 'https://www.ilpost.it/podcasts/morning/',
+        'info_dict': {
+            'id': 'morning',
+            'display_id': 'morning',
+            'title': 'Morning',
+            'series': 'Morning',
+        },
+        'playlist_mincount': 20,
+    }, {
+        'url': 'https://www.ilpost.it/podcasts/basaglia-e-i-suoi/',
+        'info_dict': {
+            'id': 'basaglia-e-i-suoi',
+            'display_id': 'basaglia-e-i-suoi',
+            'title': 'Basaglia e i suoi',
+            'series': 'Basaglia e i suoi',
+        },
+        'playlist_mincount': 5,
+    }]
+
+    def _real_extract(self, url):
+        display_id = self._match_valid_url(url).group('id')
+        data = self._download_json(f'https://api-prod.ilpost.it/podcast/v1/podcast/{display_id}?hits=20', display_id)
+
+        try:
+            podcast = data['data'][0]['parent']
+        except KeyError:
+            raise ExtractorError('Failed to extract series')
```

Comment:
- Isn't the `display_id` the same as the podcast ID / playlist ID?

- It looks like we will always max out at 20 entries. shouldn't we figure out how the API does pagination?

```suggestion
        podcast_id = self._match_id(url)
        # FIXME: This only gets 20 entries max
        data = self._download_json(
            f'https://api-prod.ilpost.it/podcast/v1/podcast/{podcast_id}',
            display_id, query={'hits': '20'})['data']
```